crates/symbolicator-native/src/symbolication/process_minidump.rs

```diff
@@ -344,21 +376,22 @@ async fn stackwalk(
     // After stackwalking, `provider.cficaches` contains entries for exactly
     // those modules that were referenced by some stack frame in the minidump.
     let mut modules = vec![];
+    let mut cficaches = provider.cficaches.into_inner().unwrap();
     for module in process_state.modules.by_addr() {
         let key = LookupKey::new(module);
 
         // Discard modules that weren't used and don't have any valid id to go by.
         let debug_id = module.debug_identifier();
         let code_id = module.code_identifier();
-        if !provider.cficaches.contains_key(&key) && debug_id.is_none() && code_id.is_none() {
+        if !cficaches.contains_key(&key) && debug_id.is_none() && code_id.is_none() {
             continue;
         }
 
         let mut obj_info = object_info_from_minidump_module(ty, module);
 
-        obj_info.unwind_status = Some(match provider.cficaches.get(&key).await {
-            None => ObjectFileStatus::Unused,
-            Some(cfi_module) => {
+        obj_info.unwind_status = Some(match cficaches.remove(&key) {
```

Comment:
Nit, and also not directly related to the content of the PR: I find a `match` inside a `Some` a bit hard to read, can we extract this? 
```suggestion
        let unwind_status = match cficaches.remove(&key) {
```